doc/developer-guide/developer-guide.rst

```diff
@@ -0,0 +1,164 @@
+.. _Developer_Guide:
+
+#########################################
+DSM Developer Guide
+#########################################
+
+The DSM is a non-configurable CSC that looks for output telemetry files from 
+the Dome Seeing Monitor UI and sends that information across SAL for recording
+in the EFD. The CSC also has two simulation modes in addition to the standard
+operational (real) mode and they will be described later. 
+
+.. _Dependencies:
+
+Dependencies
+============
+
+* `SAL <https://ts-sal.lsst.io>`_ - 5.0.0
+* ts_xml - v6.1.0
+* ts_salobj - v6.0.0
+* `aionotify <https://github.com/rbarrois/aionotify>`_ - 0.2.0
+
+.. Linking to the previous versions may also be worthwhile, depending on the CSC
+
+.. _API:
+
+DSM API
+=============================
+
+The content in this section is autogenerated from docstrings.
+
+.. The code below should insert the docstrings from the code.
+
+.. automodapi:: lsst.ts.dsm
+    :no-main-docstr:
+    :no-inheritance-diagram:
+
+
+.. _Build:
+
+Build and Test
+==============
+
+The recommended method for development is to get the T&S development Docker container
+(need doc link). Since the DSM code is not shipped with the development container, a
+clone of the code is necessary. Determine a place on your computer and get it via this
+
+.. prompt:: bash
+
+  git clone git@github.com:lsst-ts/ts_dsm.git
+
+It is recommended that the code be cloned into another directory as the potential
+of needing other repositories is possible. Change or create the appropriate branch in the clone. Remember the location housing the repository.
+
+At the time of this writing, the development container to retrieve is ``lsstts/develop-env:c0010``. Note that it is NOT compatible with SAL 5 and SALOBJ 6 out of the 
```

Comment:
This tag and all those notes about SAL 5, etc. will go out of date too quickly. Please don't do that. It's just asking for the docs to be frustratingly out of date.

If `:develop` will work, you can recommend that. Otherwise I suggest a link to somewhere that shows what versions are available and leaving it at that.

I suggest documenting requirements somewhere else (e.g. the version history or a standard requirements file, if our build system supports that).